### PR TITLE
implement gateway.spec.infrastructure

### PR DESCRIPTION
**Pull request by zirain**
part of https://github.com/envoyproxy/gateway/issues/2148, implement `gateway.spec.infrastructure`

### CODE REVIEW

**Review by zhaohuabing — DISMISSED** [review]

**Review by Xunzhuo — DISMISSED** [review]
The current additions look good

File: internal/gatewayapi/translator.go [hunk 1]
```diff
@@ -234,13 +235,21 @@ func (t *Translator) InitIRs(gateways []*GatewayContext, resources *Resources) (
 	for _, gateway := range gateways {
 		gwXdsIR := &ir.Xds{}
 		gwInfraIR := ir.NewInfra()
+		labels := infrastructureLabels(gateway.Gateway)
```
Comment (arkodg) [inline internal/gatewayapi/translator.go:238]: how does this interact with pod labels and annotations defined in https://gateway.envoyproxy.io/v0.6.0/api/extension_types/#kubernetespodspec as well as service annotations 
https://gateway.envoyproxy.io/v0.6.0/api/extension_types/#kubernetesservicespec ?

Comment (zirain) [reply internal/gatewayapi/translator.go:238]: same key in envoyproxy take first

Comment (arkodg) [reply internal/gatewayapi/translator.go:238]: is it appended ? can we had a test case for it ?

Comment (zirain) [reply internal/gatewayapi/translator.go:238]: it's a map, so.

yes, I will add a test for it.

**Review by arkodg — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]

**Pull request merged**